packages/protocol/contracts/common/Proxy.sol

```diff
@@ -138,4 +147,20 @@ contract Proxy {
     }
     emit OwnerSet(newOwner);
   }
+
+
+  /**
+   * @dev _isContract detect whether the address is 
+   *      a contract address or externally owned account (EOA)
+   * WARNING: Calling this function from a constructor will return 0
+  *           independently if the address given as parameter is a contract or EOA
+   * @return true if it is a contract address
+   */
+  function _isContract(address addr) public view returns (bool) {
```

Comment:
This can be `internal` rather than `public`, right?

Reply:
:+1: